### DISCUSSION TITLE
Keep database, but delete sounds

### DISCUSSION THREAD

**Discussion by kinteki** [mcguirepr89/BirdNET-Pi#586] [General]
Hi,
How can I keep the detected birds database? I would like to delete the sounds because it not purge the disk anyway.

**Comment by croisez** [comment]
Hi @kinteki , 
there is already a chapter in the Wiki that explains what you want to do: https://github.com/mcguirepr89/BirdNET-Pi/wiki/Backup-and-Restore-the-Database

**Reply by reefwing** [reply]
I don't think they want to back up and restore the database. 

I vaguely remember reading somewhere that the system is supposed to delete bird song recordings when the disk/SD card is 90% full, if you have the full disk setting on purge in advanced settings. Can anyone confirm that this is correct?

If so, I have the same issue as @kinteki - my disk is 95% full but not purging. I recently uninstalled and reinstalled BirdNET-Pi because the disk got to 100% and detection stopped. 

My guess is that the system runs the disk_check script periodically to achieve this functionality. Can you run this manually?

**Reply by ehpersonal38** [reply]
Yeah >95% disk usage and disk_check goes to work deleting files only from `~/BirdSongs/Extracted/By_Date/*`

disk_check.sh is called every 3 minutes in /etc/crontab. It'd be worth seeing whether this issue is related to the crontab not running, or if the disk check script has errors when you run it manually.

Also I'd like to see the output of
`du -sh ~/BirdSongs`

**Reply by reefwing** [reply]
Thanks @ehpersonal38 - this is the output from `du -sh ~/BirdSongs` and running the scripts manually.

```
BirdNET:~/BirdNET-Pi$ du -sh ~/BirdSongs
5.6G	/home/pi/BirdSongs
BirdNET:~/BirdNET-Pi$ cd scripts
BirdNET:~/BirdNET-Pi/scripts$ sh disk_usage.sh
Disk Size=14G, Used=13G, Available=911M, 94% of disk used
BirdNET:~/BirdNET-Pi/scripts$ sh disk_check.sh
+ df -h /
+ tail -n1
+ awk {print $5}
+ used=94%
disk_check.sh: 5: Bad substitution
```

**Reply by JP-MCA** [reply]
When you look inside /etc/crontab, it appears that the disk_check.sh is not meant to be called by using the sh command.

Here is the content from /etc/crontab, relative to disk_check.sh :
`*/5 * * * * birdnet /usr/local/bin/disk_check.sh >/dev/null 2>&1`

Thus, disk_check.sh must not be called by prefixing it by the sh command.
It must be called directly, because it is an executable file.

Look here what I got on my PI:
```
birdnetpi:~/BirdNET-Pi/scripts$ sh ./disk_check.sh
+ df -h /
+ tail -n1
+ awk {print $5}
+ used=93%
./disk_check.sh: 5: Bad substitution
birdnetpi:~/BirdNET-Pi/scripts$ ./disk_check.sh
++ df -h /
++ tail -n1
++ awk '{print $5}'
+ used=93%
+ '[' 93 -ge 95 ']'
+ sleep 1
+ '[' 93 -ge 95 ']'
```

You see that when I prefix the call by sh, it fails, as in your test.

So, could you retry again, but without using the sh command before the script name ?

**Reply by ehpersonal38** [reply]
Agreed with above ^

But it's also interesting that your disk is only 94% full - that is the same percentage as on my machine (with disk_check working fine), so that indicates to me that your disk_check is working and being called periodically from the crontab, no? 

Maybe this is a misunderstanding that the cutoff is 95%, not 90% as you thought earlier.

**Reply by reefwing** [reply]
Thanks @JP-MCA - yes running it without `sh` works. I learned something today!

And yes @ehpersonal38 - I think you are correct, it is working properly and the misunderstanding was on my side re: the 90% limit. Today my disk is 92%.

**Comment by kinteki** [comment]
aftre this my disk remain 100% used:

rpi4:~/BirdNET-Pi$ cd scripts
rpi4:~/BirdNET-Pi/scripts$ du -sh ~/BirdSongs
15G     /home/zoli/BirdSongs
rpi4:~/BirdNET-Pi/scripts$ disk_check.sh
++ tail -n1
++ df -h /
++ awk '{print $5}'
+ used=100%
+ '[' 100 -ge 95 ']'
+ source /etc/birdnet/birdnet.conf
++ LATITUDE=47.9245
++ LONGITUDE=21.7499
++ BIRDWEATHER_ID=ZDUi6LoJA2R4ihtG3cMCVoR9
++ CADDY_PWD=
++ ICE_PWD=birdnetpi
++ BIRDNETPI_URL=
++ RTSP_STREAM=rtsp://192.168.1.108:8080/h264_pcm.sdp
++ APPRISE_NOTIFICATION_TITLE='New BirdNET-Pi Detection'
++ APPRISE_NOTIFICATION_BODY='A $sciname $comname was just detected with a confidence of $confidence'
++ APPRISE_NOTIFY_EACH_DETECTION=0
++ APPRISE_NOTIFY_NEW_SPECIES=0
++ APPRISE_WEEKLY_REPORT=1
++ FLICKR_API_KEY=9dc79ad87de67216b34d0b9d706ca505
++ FLICKR_FILTER_EMAIL=
++ RECS_DIR=/home/zoli/BirdSongs
++ REC_CARD=default
++ PROCESSED=/home/zoli/BirdSongs/Processed
++ EXTRACTED=/home/zoli/BirdSongs/Extracted
++ OVERLAP=0.0
++ CONFIDENCE=0.7
++ SENSITIVITY=1.25
++ CHANNELS=1
++ FULL_DISK=purge
++ PRIVACY_THRESHOLD=1
++ RECORDING_LENGTH=6
++ EXTRACTION_LENGTH=3
++ AUDIOFMT=mp3
++ DATABASE_LANG=hu
++ SILENCE_UPDATE_INDICATOR=0
++ LAST_RUN=
++ THIS_RUN=
++ IDFILE=/home/zoli/BirdNET-Pi/IdentifiedSoFar.txt
++ APPRISE_NOTIFY_NEW_SPECIES_EACH_DAY=0
++ APPRISE_MINIMUM_SECONDS_BETWEEN_NOTIFICATIONS_PER_SPECIES=0
++ FREQSHIFT_TOOL=sox
++ FREQSHIFT_HI=6000
++ FREQSHIFT_LO=3000
++ FREQSHIFT_PITCH=-1500
++ HEARTBEAT_URL=
+ case $FULL_DISK in
+ echo 'Removing oldest data'
Removing oldest data
+ cd /home/zoli/BirdSongs/Extracted/By_Date/
/usr/local/bin/disk_check.sh: line 10: cd: /home/zoli/BirdSongs/Extracted/By_Date/: No such file or directory
+ curl 'localhost/views.php?view=Species%20Stats'
++ find '/home/zoli/BirdSongs/Extracted/By_Date/*' -type f
++ wc -l
find: ‘/home/zoli/BirdSongs/Extracted/By_Date/*’: No such file or directory
++ find '/home/zoli/BirdSongs/Extracted/By_Date/*' -maxdepth 0 -type d
++ wc -l
find: ‘/home/zoli/BirdSongs/Extracted/By_Date/*’: No such file or directory
/usr/local/bin/disk_check.sh: line 12: 0 / 0: division by 0 (error token is "0")
+ sleep 1
+ '[' 100 -ge 95 ']'
+ case $FULL_DISK in
+ echo 'Removing more data'
Removing more data
+ rm -rfv '/home/zoli/BirdSongs/Processed/*'
 
![image](https://user-images.githubusercontent.com/107460149/201610833-4e8f2a7d-386e-452b-98f0-00c815ff2634.png)


_The StreamData folder not purged._

**Reply by matthew73210** [reply]
> aftre this my disk remain 100% used:
> 
> rpi4:~/BirdNET-Pi$ cd scripts rpi4:~/BirdNET-Pi/scripts$ du -sh ~/BirdSongs 15G /home/zoli/BirdSongs rpi4:~/BirdNET-Pi/scripts$ disk_check.sh ++ tail -n1 ++ df -h / ++ awk '{print $5}'
> 
> * used=100%
> * '[' 100 -ge 95 ']'
> * source /etc/birdnet/birdnet.conf
>   ++ LATITUDE=47.9245
>   ++ LONGITUDE=21.7499
>   ++ BIRDWEATHER_ID=ZDUi6LoJA2R4ihtG3cMCVoR9
>   ++ CADDY_PWD=
>   ++ ICE_PWD=birdnetpi
>   ++ BIRDNETPI_URL=
>   ++ RTSP_STREAM=rtsp://192.168.1.108:8080/h264_pcm.sdp
>   ++ APPRISE_NOTIFICATION_TITLE='New BirdNET-Pi Detection'
>   ++ APPRISE_NOTIFICATION_BODY='A $sciname $comname was just detected with a confidence of $confidence'
>   ++ APPRISE_NOTIFY_EACH_DETECTION=0
>   ++ APPRISE_NOTIFY_NEW_SPECIES=0
>   ++ APPRISE_WEEKLY_REPORT=1
>   ++ FLICKR_API_KEY=9dc79ad87de67216b34d0b9d706ca505
>   ++ FLICKR_FILTER_EMAIL=
>   ++ RECS_DIR=/home/zoli/BirdSongs
>   ++ REC_CARD=default
>   ++ PROCESSED=/home/zoli/BirdSongs/Processed
>   ++ EXTRACTED=/home/zoli/BirdSongs/Extracted
>   ++ OVERLAP=0.0
>   ++ CONFIDENCE=0.7
>   ++ SENSITIVITY=1.25
>   ++ CHANNELS=1
>   ++ FULL_DISK=purge
>   ++ PRIVACY_THRESHOLD=1
>   ++ RECORDING_LENGTH=6
>   ++ EXTRACTION_LENGTH=3
>   ++ AUDIOFMT=mp3
>   ++ DATABASE_LANG=hu
>   ++ SILENCE_UPDATE_INDICATOR=0
>   ++ LAST_RUN=
>   ++ THIS_RUN=
>   ++ IDFILE=/home/zoli/BirdNET-Pi/IdentifiedSoFar.txt
>   ++ APPRISE_NOTIFY_NEW_SPECIES_EACH_DAY=0
>   ++ APPRISE_MINIMUM_SECONDS_BETWEEN_NOTIFICATIONS_PER_SPECIES=0
>   ++ FREQSHIFT_TOOL=sox
>   ++ FREQSHIFT_HI=6000
>   ++ FREQSHIFT_LO=3000
>   ++ FREQSHIFT_PITCH=-1500
>   ++ HEARTBEAT_URL=
> * case $FULL_DISK in
> * echo 'Removing oldest data'
>   Removing oldest data
> * cd /home/zoli/BirdSongs/Extracted/By_Date/
>   /usr/local/bin/disk_check.sh: line 10: cd: /home/zoli/BirdSongs/Extracted/By_Date/: No such file or directory
> * curl 'localhost/views.php?view=Species%20Stats'
>   ++ find '/home/zoli/BirdSongs/Extracted/By_Date/_' -type f
>   ++ wc -l
>   find: ‘/home/zoli/BirdSongs/Extracted/By_Date/_’: No such file or directory
>   ++ find '/home/zoli/BirdSongs/Extracted/By_Date/_' -maxdepth 0 -type d
>   ++ wc -l
>   find: ‘/home/zoli/BirdSongs/Extracted/By_Date/_’: No such file or directory
>   /usr/local/bin/disk_check.sh: line 12: 0 / 0: division by 0 (error token is "0")
> * sleep 1
> * '[' 100 -ge 95 ']'
> * case $FULL_DISK in
> * echo 'Removing more data'
>   Removing more data
> * rm -rfv '/home/zoli/BirdSongs/Processed/*'
> 
> ![image](https://user-images.githubusercontent.com/107460149/201610833-4e8f2a7d-386e-452b-98f0-00c815ff2634.png)
> 
> _The StreamData folder not purged._

Looks related to #623

**Comment by JP-MCA** [comment]
In the script disk_check.sh, there is a problem with the following line of code :

`filestodelete=$(($(find ${EXTRACTED}/By_Date/* -type f | wc -l) / $(find ${EXTRACTED}/By_Date/* -maxdepth 0 -type d | wc -l)))`

If by some weird reason, you have the ~/BirdSongs/Extracted/By_Date/ directory missing, the result of the second "find" command returns a zero value, which leads disk_check.sh script to make a division by zero. This is bad.
We should test for the existence of /By_Date/ directory before willing to use it in any command.

This doen't explain why the /By_Date/ directory has disappeared on kinteki's RPI4.

@kinteki : could you check on your RPI4 on the Web Terminal, and show us what's inside ~/BirdSongs/Extracted directory ?
Please use the following command in the web terminal:
`ls -alh ../BirdSongs/Extracted`
(please don't blame me if I explain things to you like this, because I don't know if you're comfortable in a linux terminal)

**Comment by kinteki** [comment]
@JP-MCA  here is the response to the command:
rpi4:~/BirdNET-Pi$ ls -alh ../BirdSongs/Extracted
total 12K
drwxrwxr-x 3 zoli zoli 4.0K Nov 12 06:25 .
drwxrwxr-x 6 zoli zoli 4.0K Sep 27 07:52 ..
drwxrwxr-x 2 zoli zoli 4.0K Nov 11 11:17 Charts
lrwxrwxrwx 1 zoli zoli   57 Sep 27 07:50 favicon.ico -> /home/zoli/BirdNET-Pi/scripts/homepage/images/favicon.ico
lrwxrwxrwx 1 zoli zoli   41 Sep 27 07:50 history.php -> /home/zoli/BirdNET-Pi/scripts/history.php
lrwxrwxrwx 1 zoli zoli   37 Sep 27 07:50 images -> /home/zoli/BirdNET-Pi/homepage/images
lrwxrwxrwx 1 zoli zoli   40 Sep 27 07:50 index.php -> /home/zoli/BirdNET-Pi/homepage/index.php
lrwxrwxrwx 1 zoli zoli   42 Sep 27 07:50 overview.php -> /home/zoli/BirdNET-Pi/scripts/overview.php
lrwxrwxrwx 1 zoli zoli   21 Sep 27 07:50 phpsysinfo -> /home/zoli/phpsysinfo
lrwxrwxrwx 1 zoli zoli   38 Sep 27 07:50 play.php -> /home/zoli/BirdNET-Pi/scripts/play.php
lrwxrwxrwx 1 zoli zoli   30 Sep 27 07:50 Processed -> /home/zoli/BirdSongs/Processed
lrwxrwxrwx 1 zoli zoli   29 Sep 27 07:50 scripts -> /home/zoli/BirdNET-Pi/scripts
lrwxrwxrwx 1 zoli zoli   45 Sep 27 07:50 spectrogram.php -> /home/zoli/BirdNET-Pi/scripts/spectrogram.php
-rw-rw-r-- 1 zoli zoli    0 Nov 14 09:06 spectrogram.png
lrwxrwxrwx 1 zoli zoli   37 Sep 27 07:50 static -> /home/zoli/BirdNET-Pi/homepage/static
lrwxrwxrwx 1 zoli zoli   39 Sep 27 07:50 stats.php -> /home/zoli/BirdNET-Pi/scripts/stats.php
lrwxrwxrwx 1 zoli zoli   40 Sep 27 07:50 style.css -> /home/zoli/BirdNET-Pi/homepage/style.css
lrwxrwxrwx 1 zoli zoli   51 Sep 27 07:50 todays_detections.php -> /home/zoli/BirdNET-Pi/scripts/todays_detections.php
lrwxrwxrwx 1 zoli zoli   40 Sep 27 07:50 views.php -> /home/zoli/BirdNET-Pi/homepage/views.php
lrwxrwxrwx 1 zoli zoli   47 Sep 27 07:50 weekly_report.php -> /home/zoli/BirdNET-Pi/scripts/weekly_report.php

**Reply by croisez** [reply]
Could you recreate the By_Date directory inside `../BirdSongs/Extracted`  ?
Do this with the following command:
`mkdir -p ../BirdSongs/Extracted/By_Date`
in a Web Terminal

**Reply by croisez** [reply]
and then reboot your RPI4

**Reply by kinteki** [reply]
Unfortunatly after this the birdnet analysis module stopped and didn't work at all. it was necessary to reinstall birdnet. I am eagerly waiting to see if it will work after 'hard' upgrade.
